Added character carousel page

chenga3 opened the pull request:
https://github.com/codersforcauses/foodbank/tree/s26-character-page-carousel is messy since its got code merged from other branched. I've extracted just the character carousel stuff. Ty Peter + Shuang for most of the work 🙏 

Changes include:

- API function to retrieve characters from a particular town (from Notion)
- Components for character carousel including single page display and individual carousel items
- Dynamic routing for carousel page
- Townbox links to carousel page
- Added hook that listens to window size and returns width of viewport
- Max number of characters that display on the page based on viewport width
- Characters render in a grid so each one takes same amount of space even when numCharsOnPage < maxNumCharsOnPage
- Fixed pagination issue (previously only two pages ever rendered)

See comment below for more changes made.

Closes #226 #229 #230 #236

CodeWithTheDoctor commented:
The functioning is fantastic!!! Works great.

However, styling may need a bit of work 😅 
![image](https://user-images.githubusercontent.com/22740095/154012055-1c7a48d8-2115-44d6-bf2a-790d47e4ab1c.png)

chenga3 commented:
> The functioning is fantastic!!! Works great.
> 
> However, styling may need some small changes 😅
> 
> * Make the images preserve their original aspect ratios so that they don't appear stretched
> 
> Also it would be nice to have arrows as buttons, but that can be done as a different pr.
> 
> ![image](https://user-images.githubusercontent.com/22740095/154012055-1c7a48d8-2115-44d6-bf2a-790d47e4ab1c.png)

Ok I think this is ready for merge now. Things I've done:
- Fixed issue where background was not extending to whole screen (`min-h-screen` fixed it)
- Positioned carousel centrally (previously would be too close to top for large screens, and was skewed to the left when carousel width was much lower screen width)
- Changed page toggle buttons to arrows
- When you can't go to next/prev page, greyed out unclickable button renders to ensure carousel doesn't shift due to buttons disappearing/reappearing

Still to do:
- Fix small screen display - char images look a bit too big and cramped too close to button

CodeWithTheDoctor commented on lib/hooks.ts:
```diff
@@ -0,0 +1,25 @@
+import { useState, useEffect } from 'react'
+
+/** Listens to changes in window size and returns the width. Ref: https://blog.logrocket.com/developing-responsive-layouts-with-react-hooks/ */
+const useViewport = () => {
```
Will need to use this custom hook for retrieving viewports in the Map component too. Not relevant to this pr, but thought it would be good to inform and keep in mind so we can do that later.

CodeWithTheDoctor reviewed (approved):
Looks good to be merged!
